### PR TITLE
8-ticket first commit and done with TDD

### PR DESCRIPTION
**Pull request by Hassen-Ahmed**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -248,4 +248,63 @@ describe("All requests", () => {
...
+                    const { article } = body;
+                    expect(article["votes"]).toBe(102);
+
+                    expect(article).toHaveProperty("article_id", expect.any(Number));
+                    expect(article).toHaveProperty("title", expect.any(String));
+                    expect(article).toHaveProperty("topic", expect.any(String));
+                    expect(article).toHaveProperty("author", expect.any(String));
+                    expect(article).toHaveProperty("body", expect.any(String));
+                    expect(article).toHaveProperty("created_at", expect.any(String));
+                    expect(article).toHaveProperty("votes", expect.any(Number));
```
Comment (NRMSMITH) [inline __tests__/app.test.js:267]: You've already checked this!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -248,4 +248,63 @@ describe("All requests", () => {
...
+                    expect(article).toHaveProperty("article_img_url", expect.any(String));
+                });
+        });
+
+        test("400 should be returned when article_id is NaN.", () => {
```
Comment (NRMSMITH) [inline __tests__/app.test.js:272]: maybe your description should be reflect that this could work for any id that isn't an id

**Review by NRMSMITH — APPROVED** [review]
Tiny tweaks but feel free to correct them and then merge - no need for another PR ☄️

File: __tests__/app.test.js [hunk 3]
```diff
@@ -248,4 +248,63 @@ describe("All requests", () => {
...
+        it("200 should be returned when article_id and body validation are valid", () => {
+            return request(app)
+                .patch("/api/articles/1")
+                .expect(200)
+                .send({ inc_votes: 2 })
+                .then(({ body }) => {
+                    const { article } = body;
+                    expect(article["votes"]).toBe(102);
+
+                    expect(article).toHaveProperty("article_id", expect.any(Number));
```
Comment (roisinoreilly) [inline __tests__/app.test.js:261]: we know this is article id 1

File: __tests__/app.test.js [hunk 4]
```diff
@@ -248,4 +248,63 @@ describe("All requests", () => {
...
+                .send({ inc_votes: 2 })
+                .then(({ body }) => {
+                    expect(body.msg).toBe("Invalid request!");
+                });
+        });
+        test("400 should be returned when datatype of inc_votes is not a number.", () => {
+            return request(app)
+                .patch("/api/articles/9")
+                .expect(400)
+                .send({ inc_votes: "I am STRING!" })
```
Comment (roisinoreilly) [inline __tests__/app.test.js:285]: great edge case

**Review by roisinoreilly — APPROVED** [review]
excellent error handling, really great work. PR approved!

**Comment by Hassen-Ahmed** [comment]
Thank you so much for everything!
